### DISCUSSION TITLE
Is `from_pars` useful?

### DISCUSSION THREAD

**Discussion by rozyczko** [easyscience/.github#3] [Ideas]
From a discussion with @andped10:

EasyCore objects can be instantiated:
- Directly with Descriptors/Parameters passed
- Indirectly with raw values of those descriptors and parameters passed to `.from_pars` class method.

Should we remove the `from_pars` method and allow instantiation of an EasyCore object with raw values as an option?

I.e. replace for instance:

```python
def __init__(self, power: Descriptor, amp: Parameter):
```
with
```python
def __init__(self, power: Union[Descriptor, float], amp: Union[Parameter, float]):
```

This will make the API cleaner, especially for Notebook users.

**Comment by andped10** [comment]
I would really be in favor of getting rid of the `from_pars` and `default` methods that we currently have.

My suggestion is that we try to make the changes for a single class in ERL and see if it works.  If we are happy with the result we can start to roll out the change in the code base.

**Comment by rozyczko** [comment]
Good, let's convert this into a task then.

https://github.com/EasyScience/EasyReflectometryLib/issues/132

https://github.com/EasyScience/EasyDiffractionLib/issues/105

**Comment by rozyczko** [comment]
Closing as relevant tasks have been opened and the idea agreed on.
